### PR TITLE
enhance logic to set weights; defensive approach to avoid to set weights too fast

### PR DESCRIPTION
**Pull request by eduardogr**

### CODE REVIEW

File: storage/miner/run.py [hunk 1]
```diff
@@ -64,43 +67,42 @@ def run(self):
...
-        self.last_epoch_block = self.subtensor.get_current_block()
-
+    self.last_epoch_block = self.metagraph.last_update[self.my_subnet_uid].item()
     bt.logging.info(f"Miner starting at block: {self.last_epoch_block}")
 
     # This loop maintains the miner's operations until intentionally stopped.
     bt.logging.info(f"Starting main loop")
     step = 0
     wait_factor_next_set_weights = 0
+    tempo = 360
```
Comment (ifrit98) [inline storage/miner/run.py:77]: Eventually we just use an extrinsic to get this value from chain, yeah?

**Review by ifrit98 — APPROVED** [review]
AWESOME!

File: storage/miner/set_weights.py [hunk 2]
```diff
@@ -66,17 +68,24 @@ def set_weights(
...
-            wait_for_finalization=wait_for_finalization,
-            wallet=wallet,
-            version_key=1,
-        )
-        if wandb_on:
-            wandb.log({"set_weights": 1})
+        last_updated = metagraph.last_update[uid].item()
+        current_block = subtensor.get_current_block()
+
+        if current_block - last_updated > 180:
```
Comment (ifrit98) [inline storage/miner/set_weights.py:74]: Should this be `tempo / 2`?

File: storage/miner/run.py (hunk 1 above)
Comment (eduardogr) [reply storage/miner/run.py:77]: right, this should be read dynamically aswell

File: storage/miner/set_weights.py (hunk 2 above)
Comment (eduardogr) [reply storage/miner/set_weights.py:74]: yeah it should

File: storage/miner/run.py [hunk 3]
```diff
@@ -82,32 +73,38 @@ def run(self):
...
                     presence_message_seconds_count
                     % seconds_to_wait_to_log_presence_message
                     == 0
                 ):
-                    self.current_block = self.subtensor.get_current_block()
-
-                bt.logging.info(
-                    f"Miner UID {self.my_subnet_uid} running at block {self.current_block}..."
-                )
+                    bt.logging.info(
```
Comment (ifrit98) [inline storage/miner/run.py:105]: This is nice. This way we don't spam the log messages

File: storage/miner/set_weights.py [hunk 4]
```diff
@@ -66,17 +74,24 @@ def set_weights(
...
+                uids=torch.arange(0, len(chain_weights)),
+                netuid=netuid,
+                weights=chain_weights,
+                wait_for_inclusion=wait_for_inclusion,
+                wait_for_finalization=wait_for_finalization,
+                wallet=wallet,
+                version_key=1,
+            )
+            if wandb_on:
+                wandb.log({"set_weights": 1})
```
Comment (ifrit98) [inline storage/miner/set_weights.py:91]: Could we set `1` here if `success == True` and `0` if `success == False` to indicate in the wandb logs?

**Review by ifrit98 — APPROVED** [review]
LGTM! Thanks for digging into this.

**Pull request merged**